From b4c9a93868347b5132fb35fb4c9155fad476363e Mon Sep 17 00:00:00 2001
From: Laurentiu Palcu <laurentiu.palcu@nxp.com>
Date: Tue, 11 Sep 2018 15:21:43 +0300
Subject: [PATCH 4620/5242] MLK-19470: drm: imx: dcss: enable CTXLD only if
 not already in use

commit  cb2eac7e157f21166190b327e11375fb4ab92086 from
https://source.codeaurora.org/external/imx/linux-imx.git

Currently, it may happen to enable CTXLD if it's already in use. This
can lead to unpredictable behavior (green screen can be one).

This patch will check if CTXLD is already in use, before enabling it
again.

Signed-off-by: Laurentiu Palcu <laurentiu.palcu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/imx/dcss/dcss-ctxld.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/gpu/imx/dcss/dcss-ctxld.c b/drivers/gpu/imx/dcss/dcss-ctxld.c
index 076c427..248c68c 100644
--- a/drivers/gpu/imx/dcss/dcss-ctxld.c
+++ b/drivers/gpu/imx/dcss/dcss-ctxld.c
@@ -382,7 +382,7 @@ void dcss_ctxld_kick(struct dcss_soc *dcss)
 	dcss_trace_module(TRACE_CTXLD, TRACE_KICK);
 
 	spin_lock_irqsave(&ctxld->lock, flags);
-	if (ctxld->armed) {
+	if (ctxld->armed && !ctxld->in_use) {
 		ctxld->armed = false;
 		__dcss_ctxld_enable(dcss->ctxld_priv);
 	}
-- 
1.7.9.5

